catroid/src/main/java/org/catrobat/catroid/pocketmusic/fastscroll/FastScroller.java

```diff
@@ -0,0 +1,231 @@
+		TypedArray style = context.obtainStyledAttributes(attrs, R.styleable.fastscroll__fastScroller, R.attr.fastscroll__style, 0);
+		try {
+			bubbleColor = style.getColor(R.styleable.fastscroll__fastScroller_fastscroll__bubbleColor, STYLE_NONE);
+			handleColor = style.getColor(R.styleable.fastscroll__fastScroller_fastscroll__handleColor, STYLE_NONE);
+			bubbleTextAppearance = style.getResourceId(R.styleable.fastscroll__fastScroller_fastscroll__bubbleTextAppearance, STYLE_NONE);
+		} finally {
+			style.recycle();
+		}
+		setViewProvider(new DefaultScrollerViewProvider());
+	}
+
+	public void setViewProvider(ScrollerViewProvider viewProvider) {
+		removeAllViews();
+		this.viewProvider = viewProvider;
+		viewProvider.setFastScroller(this);
+		bubble = viewProvider.provideBubbleView(this);
+		handle = viewProvider.provideHandleView(this);
+		bubbleTextView = viewProvider.provideBubbleTextView();
+		addView(bubble);
+		addView(handle);
+	}
+
+	public void setRecyclerView(RecyclerView recyclerView) {
+		this.recyclerView = recyclerView;
+
+		if (recyclerView.getAdapter() instanceof SectionTitleProvider) {
+			titleProvider = (SectionTitleProvider) recyclerView.getAdapter();
+		}
+		recyclerView.addOnScrollListener(scrollListener);
+	}
+
+	@Override
+	public void setOrientation(int orientation) {
+		scrollerOrientation = orientation;
+		super.setOrientation(orientation == HORIZONTAL ? VERTICAL : HORIZONTAL);
+	}
+
+	@Override
+	protected void onLayout(boolean changed, int l, int t, int r, int b) {
+		super.onLayout(changed, l, t, r, b);
+
+		initHandleMovement();
+		bubbleOffset = viewProvider.getBubbleOffset();
+
+		applyStyling();
+
+		if (!isInEditMode()) {
+			scrollListener.updateHandlePosition(recyclerView);
+		}
+	}
+
+	private void applyStyling() {
+		if (bubbleColor != STYLE_NONE) {
+			setBackgroundTint(bubbleTextView, bubbleColor);
+		}
+		if (handleColor != STYLE_NONE) {
+			setBackgroundTint(handle, handleColor);
+		}
+		if (bubbleTextAppearance != STYLE_NONE) {
+			TextViewCompat.setTextAppearance(bubbleTextView, bubbleTextAppearance);
+		}
+	}
+
+	private void setBackgroundTint(View view, int color) {
+		final Drawable background = DrawableCompat.wrap(view.getBackground());
+		if (background == null) {
+			return;
+		}
+		DrawableCompat.setTint(background.mutate(), color);
+		view.setBackground(background);
+	}
+
+	private void initHandleMovement() {
+		handle.setOnTouchListener(new OnTouchListener() {
+			@Override
+			public boolean onTouch(View v, MotionEvent event) {
+				requestDisallowInterceptTouchEvent(true);
+				if (event.getAction() == MotionEvent.ACTION_DOWN || event.getAction() == MotionEvent.ACTION_MOVE) {
+					if (titleProvider != null && event.getAction() == MotionEvent.ACTION_DOWN) {
+						viewProvider.onHandleGrabbed();
+					}
+					manuallyChangingPosition = true;
+					float relativePos = getRelativeTouchPosition(event);
+					setScrollerPosition(relativePos);
+					setRecyclerViewPosition(relativePos);
```

Comment:
The scroller position (the number in the bubble) refers to the right side tact in the view.
If you scroll from page 1 to page 2, the view does not change.
If a song has n tacts, it's possible to scroll until tact n+1 (which is the plus button for adding a new tact).
I would suggest to refer to the left side tact in the view to fix this issue.